samples/language/Ranges.qs

```diff
@@ -0,0 +1,49 @@
+/// # Sample
+/// Ranges
+///
+/// # Description
+/// Ranges are a way of expression a range of integer values in Q#.
+/// Range expressions can be either closed ranges with definite start
+/// and end values, such as 1..10 which contains all the integers from
+/// 1 to 10 inclusively, or they can be open ranges that have
+/// indeterminate start, end, or both. Ranges can also optionally specify
+/// a step size with an integer in the middle of the expression, such as
+/// 0..2..10, which contains all the even numbers from 0 to 10 inclusively.
+namespace MyQuantumApp {
+
+    @EntryPoint()
+    operation Main() : Unit {
+
+        // Ranges can be captured as local variables.
+        let local_range = 0..10;
+
+        // Ranges are used in for loop expressions. They must be closed
+        // ranges.
+        // Here we build an array of the squares of the numbers 0 through 10.
+        mutable array = [0, size=11];
+        for i in 0..10 {
+            set array w/= i <- i*i;
+        }
+        Message($"{array}");
```

Comment:
```suggestion
        mutable array = [];
        for i in 0..10 {
            set array += [i^2];
        }
```

Reply:
Note the use of array concatenation instead of copy and update (this is the pattern that we follow in the library), and we use it since we believe this pattern could potentially be easy to detect in the evaluator and we could fine-tune performance for it.

Note also the lack of `Message` since we aim to keep samples brief and the call to `Message` does not add any value on its own.